bootstrap/090_post_setup.sql

```diff
@@ -304,6 +304,40 @@ exception
         RAISE;
 END;
 
+-- Create the WAREHOUSE_SCHEDULING task without a schedule only if it doesn't exist.
+-- If we CREATE OR REPLACE this task, we will miss scheduling after upgrades because the
+-- previous schedule will be overwritten.
+CREATE TASK IF NOT EXISTS TASKS.WAREHOUSE_SCHEDULING_0
+    ALLOW_OVERLAPPING_EXECUTION = FALSE
+    USER_TASK_MANAGED_INITIAL_WAREHOUSE_SIZE = "X-Small"
+    AS
+    execute task TASKS.WAREHOUSE_SCHEDULING;
+
+CREATE TASK IF NOT EXISTS TASKS.WAREHOUSE_SCHEDULING_15
+    ALLOW_OVERLAPPING_EXECUTION = FALSE
+    USER_TASK_MANAGED_INITIAL_WAREHOUSE_SIZE = "X-Small"
+    AS
+    execute task TASKS.WAREHOUSE_SCHEDULING;
+
+CREATE TASK IF NOT EXISTS TASKS.WAREHOUSE_SCHEDULING_30
+    ALLOW_OVERLAPPING_EXECUTION = FALSE
+    USER_TASK_MANAGED_INITIAL_WAREHOUSE_SIZE = "X-Small"
+    AS
+    execute task TASKS.WAREHOUSE_SCHEDULING;
+
+CREATE TASK IF NOT EXISTS TASKS.WAREHOUSE_SCHEDULING_45
+    ALLOW_OVERLAPPING_EXECUTION = FALSE
+    USER_TASK_MANAGED_INITIAL_WAREHOUSE_SIZE = "X-Small"
+    AS
+    execute task TASKS.WAREHOUSE_SCHEDULING;
+
+-- The real warehouse scheduling logic cannot be in python with serverless task. Make a dummy task body.
+CREATE OR REPLACE TASK TASKS.WAREHOUSE_SCHEDULING
+    ALLOW_OVERLAPPING_EXECUTION = FALSE
+    USER_TASK_MANAGED_INITIAL_WAREHOUSE_SIZE = "X-Small"
+    as
+    select 1;
```

Comment:
I assume this will be filled in later?

Reply:
Right, this will be changed to call the (pending) SQL procedure which mimics what the python procedure does (since we can't call a python proc from a serverless task).

I realized that the scheduling of these tasks (which _can_ be in python) should be reviewed on its own, rather than trying to push everything in at the same time.